packages/interactivity/src/directives.js

```diff
@@ -242,11 +242,7 @@ export default () => {
 				if ( defaultEntry ) {
 					const { namespace, value } = defaultEntry;
 					// Check that the value is a JSON object. Send a console warning if not.
-					if (
-						typeof SCRIPT_DEBUG !== 'undefined' &&
-						SCRIPT_DEBUG === true &&
-						! isPlainObject( value )
-					) {
+					if ( globalThis.SCRIPT_DEBUG && ! isPlainObject( value ) ) {
```

Comment:
I'm really surprised that `globalThis` isn't throwing any errors but at the same time it simplifies everything:

<img width="404" alt="Screenshot 2024-05-10 at 09 20 29" src="https://github.com/WordPress/gutenberg/assets/699132/157d4463-917d-431e-9d1c-25026a2a4cbe">